src/main/java/com/bruincreates/server/service/OrderService.java

```diff
@@ -178,4 +178,13 @@ public SellerOrderResponse getSellerHistory(String username) throws BadRequestEx
         response.setCancelledOrder(cancelled);
         return response;
     }
+
+    public boolean hasCompletedOrder(String username, String productId) {
+
+        OrderExample orderExample = new OrderExample();
+        orderExample.createCriteria().andProductIdEqualTo(productId).andOrderStatusEqualTo((byte) 4).andBuyerIdEqualTo(username);
```

Comment:
Put each and statement on a separate line to improve the readability

Please avoid magic number 4, I would suggest declaring a const byte completedOrder = 4 in this class